### DISCUSSION TITLE
the yolov5 model performance runned by DJL  is poor

### DISCUSSION THREAD

**Discussion by zhangkai253** [deepjavalibrary/djl#3378] [Q&A] [answered]

**Comment by frankfliu** [comment]
Are you using our model zoo model? Our model zoo won't be able to detect Chinese word. How you convert your model?
Can you share your java code that use DJL? How to do the pre/post processing?

Can you share your model and code for both java and Python?

**Comment by frankfliu** [comment]
Most likely this is related to `YoloV5Translator`, the image processing in YoloVeTranslator is different from your python code.

You might need to check if the Pipeline is matching what you are using in the python code.

Can you share your model and python code?

**Reply by zhangkai253** [reply]
Yes, I think so. you can get the python code from the link: https://github.com/ultralytics/yolov5/blob/master/detect.py

**Reply by frankfliu** [reply]
Can you share your model? Our built-in yolov5s model zoo model seems working fine.

**Reply by zhangkai253** [reply]
Yes, I tried the built-in model ever, it worked fine indeed. Can you give me your email adress? It seems that I cant upload the model here.

**Reply by frankfliu** [reply]
Can you upload to your git repo? the file might be too big for email.

**Reply by zhangkai253** [reply]
HI Frank, any idea?

**Reply by frankfliu** [reply]
Will take a look this weekend.

**Comment by frankfliu** [comment, marked as answer]
@zhangkai253 

The current Yolov5Translator resize the image to 640x640, but `detect.py` pad the image to 640x640, this causing detection quality reduced.

You can manually pad your image to 640x640, and you will get the same result.

I created a PR to add a `CenterFit` image operation to solve this issue: https://github.com/deepjavalibrary/djl/pull/3425
Once this PR is merged you can use the follow code to get the result:

```
        Criteria<Image, DetectedObjects> criteria =
                Criteria.builder()
                        .setTypes(Image.class, DetectedObjects.class)
                        .optModelPath(Paths.get("best.pt"))
                        .optTranslatorFactory(new YoloV5TranslatorFactory())
                        .optArgument("height", 640)
                        .optArgument("width", 640)
                        .optArgument("centerFit", true)
                        .optArgument("nmsThreshold", 0.45)
                        .optArgument("threshold", 0.27)
                        .optArgument("synset", "pa,chong,wang,luo")
                        .optEngine("PyTorch")
                        .build();
```
